nomos-da/network/core/src/protocols/dispersal/validator/behaviour.rs

```diff
@@ -145,3 +145,431 @@ impl<M: MembershipHandler<Id = PeerId, NetworkId = SubnetworkId> + 'static> Netw
+        }
+    }
+
+    pub fn executor_swarm(
+        addressbook: AddressBook,
+        key: Keypair,
+        peer_id: PeerId,
+        membership: impl MembershipHandler<NetworkId = u32, Id = PeerId> + 'static,
+    ) -> libp2p::Swarm<
+        DispersalExecutorBehaviour<impl MembershipHandler<NetworkId = u32, Id = PeerId>>,
+    > {
+        libp2p::SwarmBuilder::with_existing_identity(key)
+            .with_tokio()
+            .with_other_transport(|keypair| quic::tokio::Transport::new(quic::Config::new(keypair)))
+            .unwrap()
+            .with_behaviour(|_key| {
+                DispersalExecutorBehaviour::new(peer_id, membership, addressbook)
+            })
+            .unwrap()
+            .with_swarm_config(|cfg| {
+                cfg.with_idle_connection_timeout(std::time::Duration::from_secs(u64::MAX))
+            })
+            .build()
+    }
+
+    pub fn validator_swarm(
+        key: Keypair,
+        membership: impl MembershipHandler<NetworkId = u32, Id = PeerId> + 'static,
+    ) -> libp2p::Swarm<
+        DispersalValidatorBehaviour<impl MembershipHandler<NetworkId = u32, Id = PeerId>>,
+    > {
+        libp2p::SwarmBuilder::with_existing_identity(key)
+            .with_tokio()
+            .with_other_transport(|keypair| quic::tokio::Transport::new(quic::Config::new(keypair)))
+            .unwrap()
+            .with_behaviour(|_key| DispersalValidatorBehaviour::new(membership))
+            .unwrap()
+            .with_swarm_config(|cfg| {
+                cfg.with_idle_connection_timeout(std::time::Duration::from_secs(u64::MAX))
+            })
+            .build()
+    }
+
+    fn prepare_swarm_config(
+        num_instances: usize,
+        group_id: usize,
+    ) -> Vec<(Keypair, PeerId, Multiaddr)> {
+        let mut configs = Vec::with_capacity(num_instances);
+
+        for i in 0..num_instances {
+            let keypair = identity::Keypair::generate_ed25519();
+            let peer_id = PeerId::from(keypair.public());
+            let port = 5200 + group_id * num_instances + i;
+            let addr: Multiaddr = format!("/ip4/127.0.0.1/udp/{port}/quic-v1")
+                .parse()
+                .unwrap();
+
+            configs.push((keypair, peer_id, addr));
+        }
+        configs
+    }
+
+    fn create_membership(
+        num_instances: usize,
+        subnet_id: u32,
+        peer_ids: &[PeerId],
+    ) -> Neighbourhood {
+        let mut membership = HashMap::default();
+
+        for peer_id in peer_ids.iter().take(num_instances) {
+            membership.insert(*peer_id, HashSet::from([subnet_id]));
+        }
+
+        Neighbourhood { membership }
+    }
+
+    async fn run_executor_swarm(
+        mut swarm: Swarm<
+            DispersalExecutorBehaviour<
+                impl MembershipHandler<NetworkId = u32, Id = PeerId> + Sized + 'static,
+            >,
+        >,
+        messages_to_expect: usize,
+        mut terminator_rx: watch::Receiver<bool>,
```

Comment:
Receiver can expect unit type `()`. Watcher will signal all receivers even if the value changes from unit to unit.